File: VB/RowDoubleClick_MVVM/MainViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.DataAnnotations
Imports DevExpress.Mvvm.Xpf
Imports DevExpress.Xpf.Core
Imports System.Collections.Generic
Imports System.Collections.ObjectModel
Imports System.Linq

Namespace RowDoubleClick_MVVM

    Public Class DataItem

        Public Property Id As Integer

        Public Property Name As String
    End Class

    Public Class MainViewModel
        Inherits ViewModelBase

        Public ReadOnly Property Items As ObservableCollection(Of DataItem)

        Public Sub New()
            Items = New ObservableCollection(Of DataItem)(GetItems())
        End Sub

        Private Function GetItems() As IEnumerable(Of DataItem)
            Return Enumerable.Range(0, 10).[Select](Function(i) New DataItem() With {.Id = i, .Name = "item " & i})
        End Function

        <Command>
        Public Sub RowDoubleClick(ByVal args As RowClickArgs)
            Call DXMessageBox.Show("Row double click: " & CType(args.Item, DataItem).Name)
        End Sub
    End Class
End Namespace

File: CS/RowDoubleClick_MVVM/MainViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;
using DevExpress.Mvvm.Xpf;
using DevExpress.Xpf.Core;
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Windows;

namespace RowDoubleClick_MVVM {
    public class DataItem {
        public int Id { get; set; }
        public string Name { get; set; }
    }
    public class MainViewModel : ViewModelBase {
        public ObservableCollection<DataItem> Items { get; }
        public DataItem CurrentItem {
            get { return GetProperty(() => CurrentItem); }
            set { SetProperty(() => CurrentItem, value); }
        }
        public MainViewModel() {
            Items = new ObservableCollection<DataItem>(GetItems());
        }
        IEnumerable<DataItem> GetItems() {
            return Enumerable.Range(0, 10).Select(i => new DataItem() { Id = i, Name = "item " + i });
        }

        [Command]
        public void RowDoubleClick(RowClickArgs args) {
            DXMessageBox.Show("Row double click: " + CurrentItem.Name);
        }
    }
}
